pymove/preprocessing/map_matching.py

```diff
@@ -364,62 +416,111 @@ def interpolate_add_deltatime_speed_features(
+                    count, size, start_time, curr_perc_int, step_perc=20
+                )
                 continue
 
             # compute delta_time and distance between points
             # values = (ut.shift(move_data.at[tid, "time"][filter_nodes].astype(np.float64), -1)
             # - move_data.at[tid, "time"][filter_nodes]) / 1000
             # ut.change_move_datafeature_values_using_filter(move_data, tid, "delta_time", filter_nodes, values)
-            delta_time = ((shift(y_.astype(np.float64), -1) - y_) / 1000.0)[:-1]
+            delta_time = ((shift(y_.astype(np.float64), -1) - y_) / 1000.0)[
+                :-1
+            ]
             dist_curr_to_next = (shift(x_, -1) - x_)[:-1]
             speed = (dist_curr_to_next / delta_time)[:-1]
 
             assert np.all(
-                delta_time <= max_time_between_adj_points), "delta_time between points cannot be more than {}".format(
-                max_time_between_adj_points)
+                delta_time <= max_time_between_adj_points
+            ), "delta_time between points cannot be more than {}".format(
+                max_time_between_adj_points
+            )
             assert np.all(
-                dist_curr_to_next <= max_dist_between_adj_points), "distance between points cannot be more than {}".\
-                format(max_dist_between_adj_points)
-            assert np.all(speed <= max_speed), "speed between points cannot be more than {}".format(max_speed)
+                dist_curr_to_next <= max_dist_between_adj_points
+            ), "distance between points cannot be more than {}".format(
+                max_dist_between_adj_points
+            )
+            assert np.all(
+                speed <= max_speed
+            ), "speed between points cannot be more than {}".format(max_speed)
 
-            assert np.all(x_[1:] >= x_[:-1]), "distance feature is not in ascending order"
+            assert np.all(
+                x_[1:] >= x_[:-1]
+            ), "distance feature is not in ascending order"
 
             f_intp = interp1d(x_, y_, fill_value="extrapolate")
 
-            x2_ = move_data.at[tid, "distFromTrajStartToCurrPoint"][filter_nodes]
-            assert np.all(x2_[1:] >= x2_[:-1]), "distances in nodes are not in ascending order"
+            x2_ = move_data.at[tid, "distFromTrajStartToCurrPoint"][
+                filter_nodes
+            ]
+            assert np.all(
+                x2_[1:] >= x2_[:-1]
+            ), "distances in nodes are not in ascending order"
 
             intp_result = f_intp(x2_)  # .astype(np.int64)
-            assert np.all(intp_result[1:] >= intp_result[:-1]), "resulting times are not in ascending order"
+            assert np.all(
+                intp_result[1:] >= intp_result[:-1]
+            ), "resulting times are not in ascending order"
 
-            assert ~np.isin(np.inf, intp_result), "interpolation results with np.inf value(s)"
+            assert ~np.isin(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5421338)